src/ToggleButton/ToggleButtonGroup.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import withStyles from '../styles/withStyles';
+
+export const styles = theme => ({
+  root: {
+    transition: theme.transitions.create('background,box-shadow'),
+    background: 'transparent',
+    borderRadius: 2,
+    overflow: 'hidden',
+  },
+
+  selected: {
+    background: theme.palette.background.paper,
+    boxShadow: theme.shadows[2],
+  },
+});
+
+function isValueSelected(value, candidate) {
+  if (candidate === undefined || value === undefined) {
+    return false;
+  }
+
+  if (Array.isArray(candidate)) {
+    return candidate.indexOf(value) >= 0;
+  }
+
+  return value === candidate;
+}
+
+class ToggleButtonGroup extends React.Component {
+  handleChange = (e, buttonValue) => {
+    const { onChange, value } = this.props;
+
+    if (!onChange) {
+      return;
+    }
+
+    const index = value && value.indexOf(buttonValue);
+    let newValue;
+
+    if (index >= 0) {
+      newValue = [...value];
+      newValue.splice(index, 1);
+    } else {
+      newValue = value ? [...value, buttonValue] : [buttonValue];
+    }
+
+    onChange(e, newValue);
+  };
+
+  handleExclusiveChange = (e, buttonValue) => {
+    const { onChange, value } = this.props;
+
+    if (!onChange) {
+      return;
+    }
+
+    onChange(e, value === buttonValue ? null : buttonValue);
+  };
+
+  render() {
+    const {
+      children: childrenProp,
+      classes,
+      className: classNameProp,
+      exclusive,
+      value,
+      onChange,
+      selected: selectedProp,
+      ...other
+    } = this.props;
+    let groupSelected = selectedProp;
+
+    const children = React.Children.map(childrenProp, child => {
+      if (!React.isValidElement(child)) {
+        return null;
+      }
+
+      const { selected: buttonSelected, value: buttonValue } = child.props;
+
+      const selected =
+        buttonSelected === undefined ? isValueSelected(buttonValue, value) : buttonSelected;
+
+      if (selectedProp === undefined) {
+        groupSelected = groupSelected || selected;
+      }
+
+      return React.cloneElement(child, {
+        selected,
+        onChange: exclusive ? this.handleExclusiveChange : this.handleChange,
```

Comment:
In this implementation the signature for the call to `onChange` changes slightly based on the `exclusive` prop. When exclusive `onChange` is called with a single value and when false it is called with an array of values. I don't have a really strong opinion on best approach. The adaptive implementation is really convenient for consumers of the component but the variable signature might have surprising results later. Happy to entertain other perspectives.